### PR TITLE
Pin tasks to top of kanban board

### LINKED ISSUE #385: Pin tasks to top of task list - virtual category above priority
## Summary

Allow users to pin specific tasks to the top of the task list, appearing in a new virtual category above the existing priority column.

## Desired behaviour

- A "pinned" mechanism (e.g. context menu action, frontmatter flag, or drag target) that promotes a task above all other categories
- Pinned tasks appear in a distinct section at the top of the kanban board, above the priority column
- This is a virtual/display-only category - it doesn't change the task's actual state, just its visual position
- Unpinning returns the task to its real state column

## Open questions

- How is pin state stored? Frontmatter field, plugin data, or custom order metadata?
- Should pinned tasks still show their real state as a badge/indicator?
- Limit on number of pinned tasks, or unlimited?
- Does pin order support manual reordering within the pinned section?

### PR DESCRIPTION
**Pull request by tomcorke**
## Summary

- Adds a **PinStore** that persists pinned item IDs in plugin data (`data.json`) as an ordered array, with support for reorder, rekey, and toggle operations
- Renders a virtual **"Pinned" section** above all real columns in the kanban board, with an amber header accent matching the existing section color pattern
- Pinned cards display a **real-state badge** (Priority/Active/To Do/Done) with per-column coloring so the item's actual state is never lost
- Framework injects **"Pin to Top" / "Unpin"** at the top of every card's context menu - this is a framework feature, not adapter-specific
- Full **drag-drop integration**: drag into pinned section to pin, drag out to unpin + move, reorder within pinned section
- Guards against the virtual `__pinned__` column ID leaking into adapter operations (mover, split, insert-after)
- Backward-compatible: no PinStore = no pinned section, zero behavioral change

## Design decisions

- **Storage**: Plugin data, not frontmatter - pinning is display-only and shouldn't pollute task files
- **Architecture**: PinStore is injected into ListPanel via `setPinStore()` - clean separation, no constructor changes
- **Virtual column**: Uses `__pinned__` as a column ID that never reaches the adapter's mover/parser
- **State badge**: Inserted as first child of `.wt-card-meta` row using CSS classes per state (`wt-state-badge-priority`, etc.)

## Test plan

- [x] 11 PinStore unit tests (pin/unpin/toggle/reorder/rekey/defensive copy)
- [x] 11 ListPanel integration tests (section rendering, card exclusion, state badge, isPinned context, rekey, ordering, backward compat)
- [x] All 1048 tests pass
- [x] Build clean (`pnpm run build`)
- [ ] Manual: right-click card > "Pin to Top" shows pinned section
- [ ] Manual: pinned card shows real state badge
- [ ] Manual: drag card into pinned section pins it
- [ ] Manual: drag card out of pinned section unpins it
- [ ] Manual: pinned section persists across plugin reload

Closes #385

Co-Authored-By: Claude Opus 4.6 (1M context) <noreply@anthropic.com>

### CODE REVIEW

File: styles.css [hunk 1]
```diff
@@ -1455,6 +1456,49 @@ button.wt-spawn-claude-ctx:hover svg {
   cursor: help;
 }
 
+/* =============================================================================
+   Pinned items - state badge and visual treatment
+   ============================================================================= */
+
+/* Subtle left-border accent on pinned cards */
+.wt-card-pinned {
+  border-left: 2px solid #d69e2e;
```
[the remark this answers is not in the document]
Comment (tomcorke) [reply styles.css]: Fixed in 19a33bc - switched to `box-shadow: inset 2px 0 0 #d69e2e` so the pinned accent no longer conflicts with the selection `border-left-color`.

File: src/framework/ListPanel.ts [hunk 2]
```diff
@@ -126,101 +136,180 @@ export class ListPanel {
...
-      });
+  /**
+   * Render a single section (column) of the kanban board. Used for both
+   * regular columns and the virtual pinned section.
+   */
+  private renderSection(
+    columnId: string,
+    label: string,
+    items: WorkItem[],
+    pinnedSet: Set<string>,
```
[the remark this answers is not in the document]
Comment (tomcorke) [reply src/framework/ListPanel.ts]: Fixed in 9a02de7 - removed the unused `pinnedSet` parameter from `renderSection` and its two call-site arguments.

File: src/framework/ListPanel.ts [hunk 3]
```diff
@@ -583,13 +692,60 @@ export class ListPanel {
...
+      }
+
+      if (sourceColumn === PINNED_COLUMN_ID) {
+        // Dragging FROM pinned TO a regular column: unpin + move if state differs
+        if (item && dragId && this.pinStore) {
+          await this.pinStore.unpin(dragId);
+          if (item.state !== columnId) {
+            const didMove = await this.moveToColumn(item, columnId);
+            if (!didMove) return;
+          }
```
[the remark this answers is not in the document]
Comment (tomcorke) [reply src/framework/ListPanel.ts]: Fixed in 0087775 - the move is now attempted first; unpin only happens after `moveToColumn` succeeds. A failed move returns early with the item still pinned.

File: src/core/PinStore.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+   * Returns true if the item was found and re-keyed.
+   */
+  rekey(oldId: string, newId: string): boolean {
+    const idx = this.pinnedIds.indexOf(oldId);
+    if (idx < 0) return false;
+    this.pinnedIds[idx] = newId;
+    // Persist asynchronously - caller is responsible for triggering save
+    void this.persist();
+    return true;
+  }
```
[the remark this answers is not in the document]
Comment (tomcorke) [reply src/core/PinStore.ts:90]: Fixed in c5c47d1 - `rekey` now removes any existing entry for `newId` before replacing `oldId`, preventing duplicates. Added a test for this edge case.

File: src/core/interfaces.ts [hunk 5]
```diff
@@ -175,6 +175,12 @@ export interface CardActionContext {
   onClearResumeSessions(): Promise<void>;
   /** Whether this item currently shows the framework resume indicator. */
   hasResumeSessions(): boolean;
+  /** Pin this item to the top of the kanban board. */
+  onPin(): void;
+  /** Unpin this item, returning it to its real state column. */
+  onUnpin(): void;
+  /** Whether this item is currently pinned. */
+  isPinned(): boolean;
```
[the remark this answers is not in the document]
Comment (tomcorke) [reply src/core/interfaces.ts]: Fixed in 52ae56b - made `onPin`, `onUnpin`, and `isPinned` optional on `CardActionContext`. Call sites use optional chaining. The framework still always provides them, but external adapters and tests are no longer forced to implement them.
